keras/layers/rnn/base_rnn.py

```diff
@@ -905,8 +905,20 @@ def reset_states(self, states=None):
                         self.variable_dtype or backend.floatx(),
                     )
                 )
+            if len(flat_init_state_values) == 1:
+
+                def assign_unique_var(x):
+                    return backend.variable(x, name="state")
+
+            else:
+                unique_names = iter(["hidden_state", "output_state"])
```

Comment:
Note that this is the base RNN layer, which could support all different type of RNN cell, eg (stack multiple RNN cell together or customized RNN cell with different states), so this hard code here is not correct.

Reply:
Valid point. I added a `_gen_state_names` into the LSTM cell. By default, we use `state`, `state_1`, `state_2`, ...

If a RNN cell provides `_gen_state_names`, it is used to generate the names.

Reply:
As there is no base `RNNBaseCell` (at least LSTM is not inheriting from any) I implemented it using `getattr(cell, '_gen_state_names`, fallback_implementation)`